File: nice-traffic-light/dotnet/src/NiceTrafficLight/Presenter.cs
namespace NiceTrafficLight;

public static class Presenter
{
    public static void Present(TrafficLightReport report, TextWriter sink)
    {
        sink.Write($"{LabelOf(report.Colour),-6} {report.Headline}\n");
        foreach (var trigger in report.Outcome.Triggers)
        {
            sink.Write($"  - {trigger}\n");
        }
    }

    private static string LabelOf(TrafficLightColour colour) => colour switch
    {
        TrafficLightColour.Green => "GREEN",
        TrafficLightColour.Amber => "AMBER",
        TrafficLightColour.Red => "RED",
        _ => throw new ArgumentOutOfRangeException(nameof(colour)),
    };
}

File: nice-traffic-light/python/src/nice_traffic_light/presenter.py
"""Format a :class:`TrafficLightReport` and write it to a text sink."""

from __future__ import annotations

from typing import TextIO

from .traffic_light import TrafficLightReport


def present(report: TrafficLightReport, sink: TextIO) -> None:
    sink.write(f"{report.colour.value:<6} {report.headline}\n")
    for trigger in report.outcome.triggers:
        sink.write(f"  - {trigger}\n")
